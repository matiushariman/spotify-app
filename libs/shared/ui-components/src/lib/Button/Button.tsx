import type { ButtonHTMLAttributes } from 'react';

/* eslint-disable-next-line */
export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {}

export function Button({ children }: ButtonProps) {
  return (
    <button className="bg-spotify-green text-spotify-black min-w-[142px] rounded-full px-4 py-2 text-sm font-bold uppercase transition-transform hover:scale-110">
      {children}
    </button>
  );
}

export default Button;
