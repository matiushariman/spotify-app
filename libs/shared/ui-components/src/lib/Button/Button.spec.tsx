import { render } from '@testing-library/react';

import Button from './Button';

describe('Button', () => {
  it('should render successfully', () => {
    const { container } = render(<Button />);
    expect(container).toMatchSnapshot();
  });
});
