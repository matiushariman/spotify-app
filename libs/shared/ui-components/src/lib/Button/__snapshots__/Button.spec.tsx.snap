// Vitest Snapshot v1

exports[`Button > should render successfully 1`] = `
<div>
  <button
    class="bg-spotify-green text-spotify-black min-w-[142px] rounded-full px-4 py-2 text-sm font-bold uppercase transition-transform hover:scale-110"
  />
</div>
`;
